File: app/src/main/java/moe/tlaster/openween/core/model/status/MediaModel.java
package moe.tlaster.openween.core.model.status;

import com.google.gson.Gson;
import com.google.gson.annotations.SerializedName;

/**
 * Created by Asahi on 2016/10/8.
 */

public class MediaModel {
    public MediaModel(String fid) {
        mFid = fid;
    }

    @SerializedName("bypass")
    private String mByPass = "unistore.image";
    @SerializedName("createtype")
    private String mCreateType = "localfile";
    @SerializedName("filterName")
    private String mFilterName = "";
    @SerializedName("stickerID")
    private String mStickerID = "";
    @SerializedName("fid")
    private String mFid = "";
    @SerializedName("type")
    private String mType = "pic";
    @SerializedName("filterID")
    private String mFilterID = "";
    @SerializedName("picStatus")
    private int mPicStatus = 0;

    @Override
    public String toString() {
        return "[" + new Gson().toJson(this) + "]";
    }
}

File: app/src/main/java/moe/tlaster/openween/core/model/status/MediaModel.kt
package moe.tlaster.openween.core.model.status

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.ObjectMapper


/**
 * Created by Asahi on 2016/10/8.
 */

class MediaModel(fid: String) {

    @field:JsonProperty("bypass")
    private val byPass = "unistore.image"
    @field:JsonProperty("createtype")
    private val createType = "localfile"
    @field:JsonProperty("filterName")
    private val filterName = ""
    @field:JsonProperty("stickerID")
    private val stickerID = ""
    @field:JsonProperty("fid")
    private var fid = ""
    @field:JsonProperty("type")
    private val type = "pic"
    @field:JsonProperty("filterID")
    private val filterID = ""
    @field:JsonProperty("picStatus")
    private val picStatus = 0

    init {
        this.fid = fid
    }

    override fun toString(): String {
        return "[" + ObjectMapper().writeValueAsString(this) + "]"
    }
}
